libs/common/src/ui-scripts/disableFormOnSubmit.js

```diff
@@ -0,0 +1,32 @@
+// Prevent multiple form submissions by pressing enter or clicking the submit button
+// The govukButton component has a preventDoubleClick property, but this only debounces in a one-second window,
+// and is therefore not sufficient for our needs as multiple forms take longer than a second to submit.
+
+const addDisableFormOnSubmit = () => {
+  const forms = document.querySelectorAll('form');
+  // only take the last form on the page, as it is foreseeable to have multiple forms on a page,
+  // and the 'continue' button should always be in the last form on a page.
+  const lastForm = forms[forms.length - 1];
+
+  if (!lastForm) {
+    return;
+  }
+
+  let hasSubmitted = false;
+
+  lastForm.addEventListener('submit', (e) => {
+    if (hasSubmitted) {
+      // prevent multiple form submissions by keyboard
+      e.preventDefault();
+    }
+    hasSubmitted = true;
+
+    const buttons = document.querySelectorAll('.govuk-button');
+    buttons.forEach((button) => {
+      button.setAttribute('disabled', '');
```

Comment:
Will this also mark the button style to `greyed out`, should we add appropriate class to disabled button for enhanced user experience?

Reply:
This does style it as greyed out already as the css selector targets `disabled`  (this is also standard html behaviour, but its good to know were also accessing our own style sheets
```
.govuk-button--disabled,
.govuk-button[disabled='disabled'],
.govuk-button[disabled] {
  opacity: 0.5;
}
```